### PR TITLE
Billing Tag support

### PR DESCRIPTION
**Pull request by OleksiiZubko**
* Add Billing Tag as an optional free-form tag to the OlpClientSettings
with validation.
* Add provided billing tag to the all requests

Resolves: OLPEDGE-823

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/OlpClientSettings.ts [hunk 1]
```diff
@@ -60,6 +60,12 @@ export interface OlpClientSettingsParams {
      * @see interface [[DownloadManager]]
      */
     dm?: DownloadManager;
+
+    /**
+     * Billing Tag is an optional free-form tag which is used for grouping billing records together.
+     * If supplied, it must be between 4 - 16 characters, contain only alpha/numeric ASCII characters [A-Za-z0-9].
+     */
+    billingTag?: string;
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/OlpClientSettings.ts]: Definitely not in Settings.
This is per request! Settings can be shared between clients.
This needs to be added to each request class: CatalogRequest, DataRequest, PartitionsRequest, etc. as an optional parameters.

Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/OlpClientSettings.ts]: Updated. Added to each request. Thank you

File: @here/olp-sdk-dataservice-read/lib/client/CatalogRequest.ts [hunk 2]
```diff
@@ -17,7 +17,26 @@
...
  */
 
+import { validateBillingTag } from "@here/olp-sdk-dataservice-read";
+
 /**
  * A class that prepare information for calls to get Catalog metadata from ConfigAPI.
  */
-export class CatalogRequest {}
+export class CatalogRequest {
+    private bTag?: string;
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogRequest.ts]: No acronyms, please.
Also everywhere else below.

Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/CatalogRequest.ts]: updated

**Pull request merged**